Flow2 page

nyeonaa opened the pull request:
## 🔗 References

- Issue (Link) : 

1. https://velog.io/@soulee__/TypeScript-Union-Type: 인터섹션 타입
2. https://smartdevpreneur.com/how-to-set-mui-button-size-width-height-margin-padding-and-font-size/: mui 버튼 사이즈


## ✔️ Required Self Checklist

- [X] My code follows the code style of this project.
- [X] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## ✔️ CheckList

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation

## ⏰ Time Spent

 4 hours

## 💬 Description

> Add screenshots or recordings

`Modal` : 개행, 가운데 정렬

![modal center](https://user-images.githubusercontent.com/98680476/218559561-0fef3f06-afa2-488c-87fe-00f350314513.png)


--------------------------------------------

`Main` :  약간의 반응형

![responsive](https://user-images.githubusercontent.com/98680476/218559884-1558fab0-abe8-473b-b356-c4dca435db99.png)

호버 효과 

![스크린샷_20230214_043648](https://user-images.githubusercontent.com/98680476/218561265-2b39b11b-65ee-4b79-991e-aaba8d9f0fb4.png)

------------------------------------------



## 🚧 Things to Note

`global`
- h-full 해결
------------------------------
`main`
- 모달 가운데 정렬 및 개행 수정
- 상수화 진행
- 불필요한 Box 지양
- hover 효과 적용
--------------------------------
`button`
- 버튼 사이즈에 대해 sx 로 조절 (height에 tailwind가 안 먹힘)
- Button 컴포넌트 이름 변경 및 ButtonProps 사용으로 복잡도 감소
-  flow 선택 컴포넌트에 반응형 반영
----------------------
- `survey` 페이지 진행 X landing만 수정 완료함
- 추후 글자크기에 대해서도 반응형으로 바꿔야 함
- 아직 버튼 중에 onClick을 쓰는 애가 없어서 추가 안 해놨는데 추후 추가해야할 수도 있음

## 🔃 Related PRs

-

nyeonaa committed:
feat: initial setting test

이니셜 세팅을 하였습니다. husky, tailwindcss, workflows, prettier, ESlint

nyeonaa committed:
chore(.eslintrc.js): er 에러 수정

nyeonaa committed:
fix(index.tsx): tailwindcss 적용 안됨 해결

nyeonaa committed:
feat(headerlogo): 로고가 담길 헤더 제작

nyeonaa committed:
feat(components): 메인화면의 flow 선택 컴포넌트 제작

nyeonaa committed:
feat(components): 메인 페이지의 인포메이션 컴포넌트 추가 & 메인 페이지 추가

nyeonaa committed:
feat(components): 메인 화면 모달창 추가

nyeonaa committed:
feat(components): link 기능 추가

nyeonaa committed:
feat(components, pages): flow2 페이지 추가 및 바디 생성

nyeonaa committed:
feat(component): linearProgressBar 추가

nyeonaa committed:
feat: 뒤로가기

nyeonaa committed:
feat(main branch): merge

nyeonaa committed:
feat: mui 클래스로 사용하던 거 수정, 페이지 이름 수정

nyeonaa committed:
feat: 모달창

nyeonaa committed:
build: mui provider 구축

nyeonaa committed:
feat: mui 사용(미완료)

DonghyunKim98 commented on provider/ui/ui.provider.tsx:
```diff
@@ -0,0 +1,23 @@
+  children: ReactNode;
+};
+
+const theme = createTheme({
+  typography: {
+    fontFamily: ['Roboto'].join(','),
+  },
+  components: {
+    MuiTypography: {},
+  },
+});
```
- `fontFamily` 는 아직 디자이너 분이랑 확인이 안된건죠?

nyeonaa replied:
Inter이랑 Manrope(?) 쓰셨는데, 아직 폰트 바꾸는 것까지 못했습니다..!

DonghyunKim98 committed:
feat : folder structure change

DonghyunKim98 committed:
feat : modal component 분리

DonghyunKim98 committed:
fix : fix the type error

DonghyunKim98 commented on components/LogoHeader/logo-header.component.tsx:
```diff
@@ -0,0 +1,16 @@
+import Image from 'next/image';
+import Link from 'next/link';
+
+export const LogoHeader = () => {
+  return (
+    <Box component="div" className="flex bg-[#A1C4DD] w-full h-16">
+      <Link href="./">
+        <IconButton aria-label="logo">
+          <Image src="/Team Logo (1).png" alt="로고" width="130" height="45" />
+        </IconButton>
+      </Link>
+    </Box>
+  );
+};
```
- `HeaderLogo` 에서 `LogoHeader` 로 바꿨습니다.
- 앞으로 어떤 헤더가 더 추가될지 모르는데, prefix 를 붙이는 편이 더 네이밍이 적절하다고 생각해요.
- 추후에 유지보수에 따라 `Header` 파일 하나를 놓고 props 로 분기처리 할 수도 있겠죠?

DonghyunKim98 commented on components/LinearProgressWithLabel/linear-progress-with-label.component.tsx:
```diff
@@ -0,0 +1,38 @@
+export const LinearWithValueLabel = () => {
+  const [progress, setProgress] = React.useState(10);
+
+  React.useEffect(() => {
+    const timer = setInterval(() => {
+      setProgress((prevProgress) => (prevProgress >= 100 ? 10 : prevProgress + 10));
+    }, 800);
+    return () => {
+      clearInterval(timer);
+    };
+  }, []);
```
- `React.useEffect` 말고 `useEffect` 를 바로 써주세요.

DonghyunKim98 commented on components/LinearProgressWithLabel/linear-progress-with-label.component.tsx:
```diff
@@ -0,0 +1,38 @@
+
+export const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }) => {
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'center' }}>
+      <Box sx={{ width: '100%', mr: 1 }}>
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
+      </Box>
+    </Box>
+  );
```
- 여기는 tailwind 가 적용이 안되는건가요???

nyeonaa replied:
아 이거는 mui 공식문서에 있는 거 그대로 가져와서 그렇고 tailwindcss 적용되는지 확인하도록 하겠습니다!

DonghyunKim98 commented on components/LinearProgressWithLabel/linear-progress-with-label.component.tsx:
```diff
@@ -0,0 +1,38 @@
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
```
- 상수화가 필요합니다.
- props.value 말고 value 를 한번에 가져와주세요.

DonghyunKim98 commented on components/LinearProgressWithLabel/linear-progress-with-label.component.tsx:
```diff
@@ -0,0 +1,38 @@
+import Typography from '@mui/material/Typography';
+import * as React from 'react';
+
+export const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }) => {
```
- intersection 을 type 으로 쓴걸 외부로 빼서 새로운 타입을 선언해주세요.

DonghyunKim98 commented on components/Modal/modal.component.tsx:
```diff
@@ -0,0 +1,31 @@
+  description?: string;
+}
+
+export const Modal = ({ isOpen, onClose, title, description }: ModalProps) => {
+  const handleClose = () => onClose();
+
+  return (
+    <MUIModal
+      open={isOpen}
+      onClose={handleClose}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description"
+      className="bg-zinc-50"
+    >
+      <Box className="absolute w-80 h-110 rounded-[20px]">
+        <Typography id="modal-modal-title" variant="h6" component="h2">
+          {title}
+        </Typography>
+        <Typography id="modal-modal-description" sx={{ mt: 2 }}>
+          {description}
+        </Typography>
+      </Box>
+    </MUIModal>
+  );
+};
```
- 여기는 중요하니깐 잘 봐두세요!
- 추후에 `fowardRef` + `useImperativeHandle` 을 이용해서 리팩토링을 할 수도 있습니다.
- 또한 반드시 `Modal` 을 사용해야할까 를 고민해보세요. 당장 MUI 에서 Modal 공식 페이지를 읽다보면 `Dialog`, `Menu`, `Drawer`, `PopOver` 등이 Modal 을 이용한 고차원 컴포넌트로 소개하고 있습니다.

nyeonaa replied:
헉 넵!! 다른 것도 사용해보겠습니다 👀

DonghyunKim98 commented on components/SurveyAnswer/survey-answer.component.tsx:
```diff
@@ -0,0 +1,11 @@
+export const SurveyAnswer = () => {
+  return (
+    <Box className="justify-center p-4">
+      <Box className="bg-[#FFE7A9] hover:bg-yellow-300 w-[calc(100%-48px)] h-[40px] rounded-full p-3 shadow-md text-center">
+        답변1
```
- 최대한 color 값을 그대로 적는 것을 지향하고 MUI 및 tailwind 에서 제공하는 color 값으로 맞춰주세요.
- theme 에서 설정해주시고, 만약 다른 색을 쓴다고 한다면 디자이너와 상의하세요.

DonghyunKim98 reviewed (changes requested):
landing 만 screen 으로 옮기고 조금의 변화를 줘봤습니다.

1. `component` 폴더에는 **공통 컴포넌트**만 있도록 관리해주세요.
     - 모든 스크린으로부터 하위 컴포넌트를 만들다보면 관리가 되지를 않습니다.
     - 물론 component/common 폴더와 component/page_name 으로 관리할 수 도 있으나, 지금 방향성이 제가 현업에 있을때 더 깔끔한 폴더 구성 방식이었습니다.
2. 최대한 `px`, color hex 값 등을 그대로 쓰지말고 variant 화 시켜주세요.

DonghyunKim98 committed:
feat : prepare-commit-msg

DonghyunKim98 committed:
12
::
index 0000000..068952b
--- /dev/null
+++ b/.huskyrc
@@ -0,0 +1,5 @@
+{
+    "hooks": {
+        "pre-commit": "echo 'Hello Gabia, woof!'"
+    }
+}
\ No newline at end of file

DonghyunKim98 committed:
feat : lint-stage

DonghyunKim98 committed:
commit-msg , prepare-commit-msg

DonghyunKim98 committed:
feat : delete hello world

DonghyunKim98 committed:
feat(husky): add husky files

nyeonaa committed:
style(linearprogresswithlabel): tailwind로 변경

DonghyunKim98 committed:
feat(review): review reflect

DonghyunKim98 committed:
feat(landing screen): typescript error

DonghyunKim98 committed:
feat(survey screen): delete all files

nyeonaa committed:
feat(component, landing): muibutton component

nyeonaa committed:
feat(typography): typography 일부 설정

nyeonaa committed:
feat(screen, pages): survey 초안 완성

nyeonaa committed:
fix: type 수정

nyeonaa committed:
style(screen): 랜딩 배경색 오류 수정

nyeonaa committed:
fix(modal): type check 오류 수정

description은 undefined일 수 있음 -> falsy 값 처리

nyeonaa committed:
fix: description 오류 해결

DonghyunKim98 commented on components/LogoHeader/logo-header.component.tsx:
```diff
@@ -0,0 +1,13 @@
+  return (
+    <Box component="div" className="flex bg-[#A1C4DD] w-full h-15">
+      <Link href="./">
+        <IconButton aria-label="logo"></IconButton>
```
- `<IconButton/>` 으로 변경이 가능해보입니다.

DonghyunKim98 commented on components/Modal/modal.component.tsx:
```diff
@@ -0,0 +1,49 @@
+    <MUIModal open={isOpen}>
+      <Box
+        className="text-center w-80 h-110"
+        onClick={() => {
+          setIsOpen(!isOpen);
+        }}
```
- props 에 함수를 선언하기 보다는 따로 네이밍을 붙여서 빼야 유지보수가 편하다고 생각해
```
onClick={handleClickModal}
```
- 그리고 `MUIModal` 에 onClick 이 있어야 정확하지 않나..?
- 완성된 버전을 봤는데 background 를 눌렀을때는 모달이 꺼지지를 않네. 이건 구현되야 한다고 생각해

DonghyunKim98 commented on components/Modal/modal.component.tsx:
```diff
@@ -0,0 +1,49 @@
+          setIsOpen(!isOpen);
+        }}
+      >
+        <Description description={description} />
```
- 이 부분을 굳이 Component 안에서 또다른 Component 를 선언할 필요가 있는지 모르겠어.
- 조건부 렌더링을 이용하면 되지 않을까??
```
{description && (<Box></Box/>)}
```
이런식으로

DonghyunKim98 commented on components/MuiButton/muibutton.component.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Button, Typography, Box } from '@mui/material';
+
+interface MuiButtonProps {
+  Color?: boolean;
+  tail?: string;
+  size?: undefined;
+  title?: string;
+  content?: string;
+  type?: boolean;
+  link?: string;
```
- Color 만 대문자인 이유를 모르겠어
- size, type 등은 반드시 필요한 props 이지 않니?
- size / type 같은 경우에는 string type 을 선언하는게 맞아
```
type: 'contained' | 'outlined'
```
- `MUIButton` 의 Props 의 경우에는 MUI 쪽에서 제공하는게 있지 않니??

DonghyunKim98 commented on components/MuiButton/muibutton.component.tsx:
```diff
@@ -0,0 +1,32 @@
+  return (
+    <>
+      <Button
+        color={Color ? 'primary' : 'secondary'}
+        variant={type ? 'contained' : 'outlined'}
```
- 이런식으로 사용하지 말고 default Props 를 사용하는 게 나을 것 같아.

DonghyunKim98 commented on provider/ui/ui.provider.tsx:
```diff
@@ -0,0 +1,47 @@
+  typography: {
+    fontFamily: ['Inter'].join(','),
+    h1: {
+      //질문
```
쓸데 없는 주석은 하지말자!

DonghyunKim98 commented on screen/landing/landing.screen.tsx:
```diff
@@ -0,0 +1,75 @@
+
+  return (
+    <>
+      {/* 이거 있으면 폰트 적용은 되는데 Hydration error가 뜹니다
+      <head>
+        <link rel="preconnect" href="https://fonts.googleapis.com" />
+        <link rel="preconnect" href="https://fonts.gstatic.com" />
+        <link
+          href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700;800&family=Nanum+Gothic&display=swap"
+          rel="stylesheet"
+        />
+      </head> */}
```
- 폰트 적용등은 global.css 나 pages/app.tsx 에서 적용해야 next.js 웹 어플리케이션에 적용이 되는 것으로 알아
- 렌딩페이지에서만 폰트를 사용할 것은 아니니깐!

DonghyunKim98 commented on components/MuiButton/muibutton.component.tsx:
```diff
@@ -0,0 +1,32 @@
+
+interface MuiButtonProps {
+  Color?: boolean;
+  tail?: string;
```
- 아 이거 tailwind 의 약자구나
- 그냥 className 으로 그대로 따라가자. 그래야 vscode extention 도 반영이 될거니깐

DonghyunKim98 reviewed (changes requested):
전반적으로 그래도 어느정도 감은 찾은 것 같아서 다행이야!

PR 을 조금만 더 잘 작성해줬으면 좋겠어. 링크가 중간에 날라가 버려서 이미지도 제대로 못본것도 있어 😥

nyeonaa committed:
fix(_app.tsx): font 적용 해결

nyeonaa committed:
feat(modal): 모달창 close 수정

nyeonaa committed:
style(muibutton): default props, optional, use Stack

DonghyunKim98 commented on .github/pull_request_template.md:
```diff
@@ -2,6 +2,9 @@
 
 - Notion Task Board (link):
 - Issue (Link)
+
+1.  https://nextjs.org/docs/basic-features/font-optimization : 폰트 적용
+
```
- github pr template 파일을 바꾼 이유가..?

nyeonaa replied:
앜ㅋㅋㅋ ㅠㅠㅠ 이거 pr 어떻게 쓰는지 몰라서 이렇게 쓰는건가 하구 바꿨었눈데 안돌려놨었네요 ㅠ 바꿔놓을게요

DonghyunKim98 commented on components/Modal/modal.component.tsx:
```diff
@@ -35,14 +23,18 @@ export const Modal = forwardRef<ModalRef, ModalProps>(({ description }, ref) =>
+      <Box className="text-center w-80 h-110">
+        {description && (
+          <Box className="text-center">
+            {description.split('\n').map((txt) => (
+              <>
+                {txt}
+                <br />
+              </>
+            ))}
```
- 개행을 표시할려면 이게 최선이었나요??
- typography 는 적용이 안되어 있는것 같은데!

DonghyunKim98 commented on components/MuiButton/muibutton.component.tsx:
```diff
@@ -1,30 +1,36 @@
   content?: string;
-  type?: boolean;
+  type?: 'contained' | 'outlined';
   link?: string;
```
- https://stackoverflow.com/questions/56696018/react-material-ui-typescript-inherit-props-from-button-to-add-different-var
- MUIButtonProps 라는게 있는데 이걸 그대로 가져오는게 더 낫지 않을까??

DonghyunKim98 replied:
- 버튼을 클릭했을때 onClick 으로 설정을 해야하지 않을까? 버튼을 클릭했을때 특정 링크로만 가도록만 동작하지는 않잖아.

DonghyunKim98 commented on screen/survey/survey.screen.tsx:
```diff
@@ -12,32 +12,20 @@ export const SurveyScreen = () => {
-            Color={false}
-            tail="p-5 shadow-md w-11/12 h-15 rounded-main"
+            color="secondary"
+            className="w-11/12 p-5 shadow-md h-15 rounded-main"
```
- rounded, width 등은 size 로 관리되야하는 props 라고 생각은 하는데, 이를 반영하기 어렵다면 일단 넘어가자

nyeonaa committed:
fix(package.json): 알 수 없는 오류 수정

nyeonaa committed:
style: pR 템플릿 잘못 고친 거 수정

nyeonaa committed:
style(modal): 개행 수정

DonghyunKim98 committed:
feat(modal): parser

nyeonaa committed:
feat(_app.tsx): height 100% 해결

nyeonaa committed:
feat(button): 버튼 컴포넌트 수정

nyeonaa committed:
feat: 모달 개행

DonghyunKim98 commented on screen/landing/landing.screen.tsx:
```diff
@@ -21,40 +20,52 @@ export const LandingScreen = () => {
+            {Intro}
             <Typography component="span" variant="h2_extrabold" className="pl-1 text-primary">
-              머물래
+              {Mml}
```
- 변수명이 정확히 무엇인지 의미가 살지를 않는데,
```
const TitleText = '머물러에 오신 것을 환영합니다.'
```
등으로 가는게 좋지 않을까? (suffix 로 Text 를 붙이자는거야)

DonghyunKim98 commented on screen/landing/landing.screen.tsx:
```diff
@@ -21,40 +20,52 @@ export const LandingScreen = () => {
+          justifyItems="center"
+        >
+          <MuiButton
+            className="w-11/12 p-5 shadow-md md:w-5/12 md:h-40 rounded-main hover:bg-secondary/50"
```
- hover 에 대한 내용은 모든 버튼에 공통적으로 사용될거라서 `Button` 에 종속되는게 좋을것같아

DonghyunKim98 commented on screen/landing/landing.screen.tsx:
```diff
@@ -21,40 +20,52 @@ export const LandingScreen = () => {
+          <MuiButton
+            className="w-11/12 p-5 shadow-md md:w-5/12 md:h-40 rounded-main hover:bg-secondary/50"
+            title={Flow1}
+            href="./survey"
+            sx={{ height: 100 }}
```
- `href` 말고 onClick 에 대한 method 를 넘겨줘야지! button 을 클릭했을때 항상 어떤 링크로 넘어가는게 아니니깐
- `sx` 에 `className` 도 있으니 정신이 없다는 느낌이 드는데, 설명이 필요할 것 같아.

DonghyunKim98 commented on provider/ui/ui.provider.tsx:
```diff
@@ -22,6 +27,12 @@ declare module '@mui/material/styles' {
     answer_regular?: React.CSSProperties;
     next_bold?: React.CSSProperties;
   }
+  interface Palette {
+    button: Palette['primary'];
+  }
+  interface PaletteOptions {
+    button: PaletteOptions['primary'];
+  }
```
- 이거 color 에서 primary 가 바로 안잡혀??

DonghyunKim98 commented on components/Button/button.component.tsx:
```diff
@@ -0,0 +1,33 @@
+        href={href}
+        sx={sx}
+      >
+        <Box className="text-center">{title && <Box typography={typography}>{title}</Box>}</Box>
```
- 개행이 좀 필요해 보인다!

DonghyunKim98 committed:
feat(landingscreen + modal): 불필요한 props 제거, 더 알아보기 쉬운 코드 스타일, MUI Button 에 맞는 코드 작성

DonghyunKim98 committed:
feat(ui-provider): add breakpoint
